### PR TITLE
Allow disabling the automatic use of our internal swtfb client by setting an env

### PR DESCRIPTION
**Pull request by LinusCDE**
There was an issue raised at #117 where presumably a new library is used as a shim for rM 2 Framebuffer support, but it doesn't seem to use the offical server, which our client relies on.

In general there had been issues raised that issues could arise even with the official shim if it gets updated but our implementation still does the old behaviour. Such applications would require to be recompiled to use `Framebuffer::device` instead of `Framebuffer::new` or otherwise not work.

This PR allows people to make libremarkable choose the old implementation in case the developer didn't care about it anyway / left the choice up to libremarkable.

It is done by setting the lengthy env `LIBREMARKABLE_FB_DISFAVOR_INTERNAL_RM2FB` to `1`, `yes` or `true` (case-insensitive). Feel free to suggest a better name, I couldn't think of any.

Package maintainers could also fix arising issues by setting that env in some wrapper script for that binary without the need of any source code change or even re-complication if the binary is recent enough.

Basically this should future proof applications built using libremarkable better.

### CODE REVIEW

File: src/framebuffer/core.rs [hunk 1]
```diff
@@ -50,7 +50,19 @@ impl Framebuffer {
         let device = &*device::CURRENT_DEVICE;
         match device.model {
             Model::Gen1 => Framebuffer::device(device.get_framebuffer_path()),
-            Model::Gen2 => Framebuffer::rm2fb(device.get_framebuffer_path()),
+            Model::Gen2 => {
+                // Auto-select old method still if env LIBREMARKABLE_FB_DISFAVOR_INTERNAL_RM2FB is set affirmatively
```
Comment (fenollp) [inline src/framebuffer/core.rs:54]: Could you add a mention in the README about this behavior plz?

File: src/framebuffer/core.rs [hunk 2]
```diff
@@ -50,7 +50,19 @@ impl Framebuffer {
...
+                if let Ok(env_answer) = std::env::var("LIBREMARKABLE_FB_DISFAVOR_INTERNAL_RM2FB") {
+                    let env_answer = env_answer.to_lowercase();
+                    if env_answer == "1" || env_answer == "true" || env_answer == "yes" {
+                        Framebuffer::device(device.get_framebuffer_path())
+                    } else {
+                        Framebuffer::rm2fb(device.get_framebuffer_path())
+                    }
+                } else {
+                    Framebuffer::rm2fb(device.get_framebuffer_path())
+                }
```
Comment (fenollp) [inline src/framebuffer/core.rs:64]: How about this? (untested)
```suggestion
                match std::env::var("LIBREMARKABLE_FB_DISFAVOR_INTERNAL_RM2FB").ok() {
                    Some("1") => Framebuffer::device(device.get_framebuffer_path()),
                    _ => Framebuffer::rm2fb(device.get_framebuffer_path()),
                }
```

Comment (fenollp) [reply src/framebuffer/core.rs:64]: (I'm more inclined to be restrictive in the possible values, but feel free to revert that)

**Comment by Havner** [comment]
Hi,

I can't get this to work as intended. Seems that libremarkable doesn't like rm2fb client. I've compile the demo and it works properly (through built-in client). But with this it doesn't work. Has anybody managed to make it work?

(Software 2.15, original rm2fb from toltec, not timower)

```
reMarkable: /opt/demo/ LD_PRELOAD=/opt/lib/librm2fb_client.so.1 LIBREMARKABLE_FB_DISFAVOR_INTERNAL_RM2FB=1 ./demo
thread 'main' panicked at src/framebuffer/core.rs:206:9:
FBIOGET_VSCREENINFO failed
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```
reMarkable: /opt/demo/ LIBREMARKABLE_FB_DISFAVOR_INTERNAL_RM2FB=1 /opt/bin/rm2fb-client ./demo
thread 'main' panicked at src/framebuffer/core.rs:206:9:
FBIOGET_VSCREENINFO failed
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
